infra_monitored_account/infra_monitored_account/infra_monitored_stack.py

```diff
@@ -82,3 +85,52 @@ def create_event_rules(self, cross_account_bus_role, cross_account_event_bus_arn
         step_functions_alerting_event_rule.add_target(rule_target)
 
         return [glue_alerting_event_rule, step_functions_alerting_event_rule]
+
+    def create_metrics_extract_iam_role(self):
+        """
+        Creates an IAM Role allowing Tooling Account's MetricsExtractor Lambda get required data from Monitored account
+        """ 
+        # todo: do we need multiple arns (lambdas)?
+        # todo: how to get tooling extract lambda role -> through AWSNaming?
+        tooling_extract_lambda_role_name = AWSNaming.IAMRole(self, CDKResourceNames.IAMROLE_EXTRACT_METRICS_LAMBDA)
+        principal_arn = AWSNaming.Arn_IAMRole(self, self.tooling_account_id, tooling_extract_lambda_role_name)
+        
+        # todo rename
+        cross_account_iam_role_extract_metrics = iam.Role(
+            self, "MonitoredAccExtractMetricsRole",
+            role_name=AWSNaming.IAMRole(self, CDKResourceNames.IAMROLE_MONITORED_ACC_EXTRACT_METRICS),            
+            assumed_by=iam.ArnPrincipal(principal_arn)
+        )
+
+        # Glue Policy
```

Comment:
Not going to add Step Functions policy for now?

Reply:
Yes, I wanted to speed up this PR finalization (as it touches all the CDK stacks and creates potential git conflict)
and defer it till the moment we define metrics required for Step Functions.